saleor/discount/utils.py

```diff
@@ -1002,8 +1002,15 @@ def _get_defaults_for_gift_line(
             "currency": order_or_checkout.currency,
         }
     else:
+        variant = (
+            ProductVariant.objects.filter(id=variant_id)
+            .select_related("product")
+            .first()
+        )
```

Comment:
Instead of fetching variant from the DB again we should go up and probably pass `gift_listing` as an argument